itez/users/api/serializers.py

```diff
@@ -73,11 +67,42 @@ def update(self, instance, validated_data):
         return instance
 
 
-class GroupModelSerializer(serializers.ModelSerializer):
-    """
-    Serializer for the Django Group model.
-    """
+class UserSerializer(serializers.ModelSerializer):
+    password = serializers.CharField(write_only=True)
+    roles = ListField(required=False, default=[], write_only=True)
+    assigned_roles = serializers.SerializerMethodField(read_only=True)
     class Meta:
-        model = Group
-        fields = '__all__'
+        model = User
+        fields = ["email", "username", "name", "password", "roles", "assigned_roles"]
         depth = 2
+    
+    def get_assigned_roles(self, obj):
+        return [group.name for group in obj.groups.all()]
```

Comment:
`return [group.name for group in obj.groups.all()]`
maybe  we say
`user = User.objects.get(user.id = request.user.id)` # get current  logged in user
`return [group.name for group in user.groups.all() ]`

Reply:
I have renamed `obj` to `user`. There is no need to query the user because it's already in the request context.